components/builder-jobsrv/src/server/mod.rs

```diff
@@ -183,34 +195,68 @@ pub async fn run(config: Config) -> Result<()> {
     let log_dir = LogDirectory::new(&config.log_dir);
     LogIngester::start(&config, log_dir, datastore.clone())?;
 
-    WorkerMgr::start(&config, &datastore, db_pool.clone())?;
-    ScheduleMgr::start(&config, &datastore, db_pool.clone())?;
+    if feat::is_enabled(feat::NewScheduler) {
+        let scheduler_datastore = SchedulerDataStoreDb::new(datastore.clone());
+        let (scheduler, scheduler_handle) = Scheduler::start(Box::new(scheduler_datastore), 1);
+        let scheduler_for_http = scheduler.clone();
 
-    info!("builder-jobsrv listening on {}:{}",
-          cfg.listen_addr(),
-          cfg.listen_port());
+        WorkerMgr::start(&config, &datastore, db_pool.clone(), Some(scheduler))?;
+
+        let http_serv = HttpServer::new(move || {
+                            let app_state = AppState::new(&config,
+                                                          &datastore,
+                                                          db_pool.clone(),
+                                                          &graph_arc,
+                                                          Some(&scheduler_for_http));
+
+                            App::new().data(JsonConfig::default().limit(MAX_JSON_PAYLOAD))
+                    .data(app_state)
+                    .wrap(Logger::default().exclude("/status"))
+                    .service(web::resource("/status").route(web::get().to(status))
+                                                    .route(web::head().to(status)))
+                    .route("/rpc", web::post().to(handle_rpc))
+                        }).workers(cfg.handler_count())
+                          .keep_alive(cfg.http.keep_alive)
+                          .bind(cfg.http.clone())
+                          .unwrap()
+                          .run();
 
-    HttpServer::new(move || {
-        let app_state = AppState::new(&config, &datastore, db_pool.clone(), &graph_arc);
-
-        App::new().data(JsonConfig::default().limit(MAX_JSON_PAYLOAD))
-                  .data(app_state)
-                  .wrap(Logger::default().exclude("/status"))
-                  .service(web::resource("/status").route(web::get().to(status))
-                                                   .route(web::head().to(status)))
-                  .route("/rpc", web::post().to(handle_rpc))
-    }).workers(cfg.handler_count())
-      .keep_alive(cfg.http.keep_alive)
-      .bind(cfg.http.clone())
-      .unwrap()
-      .run()
-      .await
-      .map_err(Error::from)
+        // This is not what we want.  try_join! would be more appropriate so that we shut down if
+        // any of the things we're joining returns an error. However, the http_serv and
+        // scheduler_handle have different error types and I can't figure out how to resolve
+        // that scenario. We also need to handle any errors on the scheduler
+        let (http_res, _sched_res) = tokio::join!(http_serv, scheduler_handle);
```

Comment:
:thinking: There may be some kind of wrapping / error mapping we can do to make this happen....